control-plane/pkg/reconciler/brokeryolo/broker.go

```diff
@@ -0,0 +1,294 @@
+	logging.FromContext(ctx).Infof("YOLO FinalizeKind (wrapper)")
+	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		return r.finalizeKind(ctx, broker)
+	})
+}
+
+func (r *Reconciler) finalizeKind(ctx context.Context, broker *eventing.Broker) reconciler.Event {
+	logging.FromContext(ctx).Infof("YOLO finalizeKind")
+
+	br := r.createReconcilerForBrokerInstance(broker)
+	result := br.DoFinalizeKind(ctx, broker)
+
+	r.IPsLister.Unregister(types.NamespacedName{Namespace: broker.Namespace, Name: broker.Name})
+
+	return result
+}
+
+func (r *Reconciler) createReconcilerForBrokerInstance(broker *eventing.Broker) *brokerreconciler.Reconciler {
+
+	return &brokerreconciler.Reconciler{
+		Reconciler: &base.Reconciler{
+			KubeClient:             r.Reconciler.KubeClient,
+			PodLister:              r.Reconciler.PodLister,
+			SecretLister:           r.Reconciler.SecretLister,
+			SecretTracker:          r.Reconciler.SecretTracker,
+			ConfigMapTracker:       r.Reconciler.ConfigMapTracker,
+			DataPlaneConfigMapName: r.Reconciler.DataPlaneConfigMapName,
+			DataPlaneConfigFormat:  r.Reconciler.DataPlaneConfigFormat,
+			DispatcherLabel:        r.Reconciler.DispatcherLabel,
+			ReceiverLabel:          r.Reconciler.ReceiverLabel,
+
+			DataPlaneNamespace:          broker.Namespace,
+			DataPlaneConfigMapNamespace: broker.Namespace,
+		},
+		Env:                        r.Env,
+		Resolver:                   r.Resolver,
+		ConfigMapLister:            r.ConfigMapLister,
+		NewKafkaClusterAdminClient: r.NewKafkaClusterAdminClient,
+		BootstrapServers:           r.BootstrapServers,
+		Prober:                     r.Prober,
+	}
+}
+
+// getManifest returns the manifest that is transformed from the BaseDataPlaneManifest with the changes
+// for the given broker
+func (r *Reconciler) getManifest(broker *eventing.Broker) (mf.Manifest, error) {
+	manifest := r.BaseDataPlaneManifest
+
+	manifest, err := manifest.Transform(mf.InjectNamespace(broker.Namespace))
+	if err != nil {
+		return mf.Manifest{}, fmt.Errorf("unable to transform base dataplane manifest with namespace injection. namespace: %s, error: %v", broker.Namespace, err)
+	}
+
+	manifest, err = manifest.Transform(appendNewOwnerRefsToPersisted(manifest.Client, broker))
+	if err != nil {
+		return mf.Manifest{}, err
+	}
+
+	if r.Env.DispatcherImage == "" {
+		return mf.Manifest{}, fmt.Errorf("unable to find DispatcherImage env var specified with 'BROKER_DISPATCHER_IMAGE'")
+	}
+	if r.Env.ReceiverImage == "" {
+		return mf.Manifest{}, fmt.Errorf("unable to find DispatcherImage env var specified with 'BROKER_RECEIVER_IMAGE'")
+	}
+
+	// replace the ${KNATIVE_KAFKA_DISPATCHER_IMAGE} string in dataplane manifest YAML
+	// with the value of KAFKA_DISPATCHER_IMAGE
+	return manifest.Transform(setImagesForDeployments(map[string]string{
+		"${KNATIVE_KAFKA_DISPATCHER_IMAGE}": r.Env.DispatcherImage,
+		"${KNATIVE_KAFKA_RECEIVER_IMAGE}":   r.Env.ReceiverImage,
+	}))
+}
```

Comment:
Can we do this once only?

Reply:
oh, yeah, good point.
The image transform can be done once, unlike other transforms.

Reply:
Only the image transform though. Injecting owner and namespace needs to be done per broker instance